discord_bots/cogs/categories.py

```diff
@@ -194,3 +228,35 @@ async def setmingamesforleaderboard(
         await self.send_success_message(
             f"The minimum number of games required to appear on the leaderboard for category **{category.name}** is now **{min_num_games}**"
         )
+
+    @command()
+    @check(is_admin)
+    async def setcategorysigmadecay(
+        self,
+        ctx: Context,
+        name: str,
+        sigma_decay_amount: float,
+        sigma_decay_grace_days: int,
+        sigma_decay_max_decay_proportion: float,
+    ) -> None:
+        """
+        Set the amount of sigma decay per day for a given category
+        """
+        session = ctx.session
+        try:
+            category: Category = (
+                session.query(Category).filter(Category.name.ilike(name)).one()
+            )
```

Comment:
This can also raise a `sqlalchemy.orm.exc.MultipleResultsFound` exception, so either catch that as well and log it, or use `.first()` instead and check if the return result is `None` -- while `.first()` would ignore the case where multiple categories exist with the same name (which is possible), at least it doesn't break the functionality of the bot (for the most part). I don't really mind what you pick either way, since this is an edge case and the best solution would be to restrict categories from having the same name, which is outside the scope of this PR.

Reply:
This is an issue in all of the category commands at present I'm pretty sure. IMO we should probably add a uniqueness constraint on the name in the db. But I agree that ought to be a separate PR. I can set it to first() for now, but it won't _work_ any better with duplicate names it just won't throw any more.